python/distributed-ucxx/distributed_ucxx/ucxx.py

```diff
@@ -279,8 +353,12 @@ def __init__(  # type: ignore[no-untyped-def]
         else:
             self._has_close_callback = False
 
+        _register_dask_resource(hash(self))
+
         logger.debug("UCX.__init__ %s", self)
 
+        weakref.finalize(self, _deregister_dask_resource, hash(self))
```

Comment:
*question*: It seems like this hash of `self` is merely used as a key so that `_deregister_dask_resource` is only ever called once per UCXX comm object.

Since there is a (probably low) potential for hash collisions, would it make sense for instead `_register_dask_resource` to return a sequence number to be used in deregister:

```python

def make_register():
    count = itertools.count()
    def register() -> int:
        tag = next(count)
        ctx._resources.add(tag)
        ...
        return tag
    return register

_register_dask_resource = make_register()

...

self.tag = _register_dask_resource()

weakref.finalize(self, _deregister_dask_resource, self.tag)
...
```

WDYT?